### DISCUSSION TITLE
How do I get the component data in the module?

### DISCUSSION THREAD

**Discussion by korenevskiy** [joomla/joomla-cms#32610] [Q&A]
Very respected gurus give me advice.
Created the **`mod_mymodule`** module.
I wrote the code in the module.

```
JFactory::getApplication()->registerEvent('onContentPrepare', [new mod_mymoduleHelper,'event']);
```

But the function returns empty attribute values.
 
I need to have the article data shown in the category blog, and then output the result to the module position.
 
1. In what order is the rendering of modules, components, and templates performed?
2. How do I get in the module the data of each blog article of the category displayed on the screen.
Since each article has a custom field with an order in the form of a product card. The module needs to know about which field each article has.

**Comment by Fedik** [comment]
The order:
 component,
 template
 modules
 template final rendering

For a module, look for example in core modules, to get an idea how they load the data, eg `mod_articles_category` https://github.com/joomla/joomla-cms/blob/4.0-dev/modules/mod_articles_category/mod_articles_category.php
`mod_articles_latest`, `mod_articles_news`, `mod_articles_popular` and much more

**Reply by Fedik** [reply]
If you want to alter the Article detail view, then you need a plugin, not a module
